autospec/tarball.py

```diff
@@ -530,6 +485,24 @@ def process_archives(archives):
             call(move_cmd)
 
 
+def process_go_archives():
+    """Set up extra archives required by go packages."""
+    global go_archives
```

Comment:
So I was really trying to avoid more globals in the other implementation. I realize this makes some things cleaner but I think globals tend to introduce more problems with understanding then it avoids once we reach a level of complexity (and tarbal.py is definitely there for me). It also potentially requires setting of global state between tests.

Reply:
Agree and tbh, I only used it as a temp solution while I'm finishing the design of a new class that would do a better management of sources. Anyway, I think I can find a way to remove the global, if really needed.

Reply:
I can live with this being intentionally temporary since you have a larger change in the works.